### PR TITLE
Fix #5, Fix#6

### LINKED ISSUE #5: App/AppRelease deletion authorizations
Currently we've got an app owner and app collaborators. Both are allowed to publish and delete specific apps and their releases. Are we fine with this? If so we can close this ticket :)

@LukasReschke @adsworth @jancborchardt

### PR DESCRIPTION
**Pull request by BernhardPosselt**
@adsworth @LukasReschke

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -101,3 +102,44 @@ This is the route that will return all releases to display inside Nextcloud's ap
...
+* **Authentication**: Basic
+
+* **Returns**:
+  * **HTTP 204**: If the app was deleted successfully
+  * **HTTP 400**: If the app could not be found
+  * **HTTP 401**: If the user is not authenticated
+  * **HTTP 403**: If the user is not authorized to delete the app
+
+## Delete An App Release
+Only an app owner or co-maintainers are allowed to delete an app release. The owner is set to the first user that pushes a release to the store, co-maintainers can be added.
```
Comment (adsworth) [inline docs/api.md]: This should read "Only app owners or co maintainers are allowed..."

and how about "Owner is the user that pushes the first release of an app to the store,..."

File: docs/api.md [hunk 2]
```diff
@@ -101,3 +102,44 @@ This is the route that will return all releases to display inside Nextcloud's ap
...
+
+
+* **Url**: DELETE /api/v1/apps/{app-id}/{app-version}
+  * **Parameters**:
+    * **app-id**: app's id, lower case ASCII characters and underscores are allowed
```
Comment (adsworth) [inline docs/api.md]: I'm pretty sure apps is correct in this case.

Comment (BernhardPosselt) [inline docs/api.md]: yes, app id would work

File: docs/api.md (hunk 1 above)
Comment (BernhardPosselt) [inline docs/api.md]: Right, will adjust it.

File: nextcloudappstore/core/models.py [hunk 3]
```diff
@@ -47,8 +47,11 @@ class Meta:
     def __str__(self):
         return self.name
 
-    def allows_editing(self, user):
```
Comment (adsworth) [inline nextcloudappstore/core/models.py]: Django permissons uses can_*. allows_* suggest that the user allows someone to edit. It doesn't sound like a check if edting is allowed.

Comment (BernhardPosselt) [inline nextcloudappstore/core/models.py]: good point, can_edit and can_delete sounds much better

**Comment by BernhardPosselt** [comment]
Issues are fixed, however github does not show the newest diff which is weird.

**Pull request merged**